js/compressionLib/index.js

```diff
@@ -0,0 +1,85 @@
+const electron = require('electron');
+const child = require('child_process');
+const app = electron.app;
+const path = require('path');
+const userData = path.join(app.getPath('userData'));
+const isMac = require('../utils/misc.js').isMac;
+const isDevEnv = require('../utils/misc.js').isDevEnv;
+const DATA_FOLDER_PATH = isDevEnv ? path.join(__dirname, '..', '..') : userData;
+const execPath = path.dirname(app.getPath('exe'));
+
+// lz4 library path
+const libraryFolderPath = isDevEnv ? path.join(__dirname, '..', '..', 'library') : path.join(execPath, 'library');
+const winArchPath = process.arch === 'ia32' ? 'lz4-win-x86.exe' : 'lz4-win-x64.exe';
```

Comment:
We are still unsure about supporting 32-bit for Electron. This may not be required for now.